### ISSUE TITLE
API Resource Not Found

### ISSUE THREAD

**Issue by Joe-Palmer** [rcorral/com_api#4]
Firstly, thank you for this extension, it looks great! However, I have been unable to get it to work yet.

To start with, I couldn't get a clean output without adding the tmpl and format parameters. This is the working URL I have:

http://www.xxxxxxx.com/index.php?tmpl=component&format=raw&option=com_api&token=81e909a38be1e9eb8cb835a06a788f4e6fab60c5&app=users&resrouce=user

But I can't get past the following response:

{"error":true,"code":404,"message":"API Resource Not Found"}

I have installed and published all of your plugins but this message is as far as I can get.

Do you have any idea what the problem is? What am I doing wrong?

Many thanks,

Joe

**Comment by rcorral** [comment]
Thank you, although I didn't originally build it I have done a decent amount of work to it.

the error "API Resource Not Found" is normally because the plugin is not enabled. It is getting past the authentication of the token so that's good, it just can't find the resource. I'm sure that you have the api_users plugin installed an enabled right?

That error can happen in both of these places in the code:
./code/components/com_api/libraries/plugin.php:                 throw new Exception( JText::_('COM_API_PLUGIN_METHOD_NOT_FOUND'), 404 );
./code/components/com_api/libraries/resource.php:                       throw new Exception( JText::_( 'COM_API_PLUGIN_METHOD_NOT_FOUND' ), 404 );

It is more than likely getting triggered on the resource.php file if you have to users plugin enabled. Make sure that there is a file inside of that plugin under the resources folder called 'user'.

Rafael

**Comment by Joe-Palmer** [comment]
Thanks for your response, it helped me work out the problem. Looking at the URL I posted carefully, resource is spelt wrong (resrouce) which I copied and pasted from your "Usage example" in README.md. A simple fix! :)

You also have the following line in "If you want to make it better" which I guess can now be removed:

I also plan on porting it to J! 2.5 when I get a chance, so any help with that would be appreciated.

Thanks for you help,

Joe
